arcade/examples/tetris.py

```diff
@@ -42,132 +41,208 @@
 
 # Define the shapes of the single parts
 tetris_shapes = [
-    [[1, 1, 1],
+    # From https://www.joe.co.uk/gaming/tetris-block-names-221127
+    [[1, 1, 1],         # Teewee
      [0, 1, 0]],
 
-    [[0, 2, 2],
+    [[0, 2, 2],         # RhodeIslandZ
      [2, 2, 0]],
 
-    [[3, 3, 0],
+    [[3, 3, 0],         # ClevelandZ
      [0, 3, 3]],
 
-    [[4, 0, 0],
+    [[4, 0, 0],         # Blue Ricky
      [4, 4, 4]],
 
-    [[0, 0, 5],
+    [[0, 0, 5],         # Orange Ricky
      [5, 5, 5]],
 
-    [[6, 6, 6, 6]],
+    [[6, 6, 6, 6]],     # Hero
 
-    [[7, 7],
+    [[7, 7],            # Smashboy
      [7, 7]]
 ]
 
 
-def create_textures():
-    """ Create a list of images for sprites based on the global colors. """
-    new_textures = []
-    for color in colors:
-        image = PIL.Image.new('RGBA', (WIDTH, HEIGHT), color)
-        new_textures.append(arcade.Texture(image))
-    return new_textures
-
-
-texture_list = create_textures()
+def rotate_counter_clockwise(shape):
+    """ Rotates a matrix counter clockwise """
+    return [[
+        shape[y][x] for y in range(len(shape))]
+            for x in range(len(shape[0]) - 1, -1, -1)]
 
 
-def rotate_counterclockwise(shape):
+def rotate_clockwise(shape):
     """ Rotates a matrix clockwise """
-    return [[shape[y][x] for y in range(len(shape))]
-            for x in range(len(shape[0]) - 1, -1, -1)]
+    return [list(l[::-1]) for l in zip(*shape)]
 
 
-def check_collision(board, shape, offset):
-    """
-    See if the matrix stored in the shape will intersect anything
-    on the board based on the offset. Offset is an (x, y) coordinate.
-    """
-    off_x, off_y = offset
-    for cy, row in enumerate(shape):
-        for cx, cell in enumerate(row):
-            if cell and board[cy + off_y][cx + off_x]:
-                return True
-    return False
+def create_texture(color):
+    """ Create a texture for sprites based on the global colors. """
+    # noinspection PyUnresolvedReferences
+    image = PIL.Image.new('RGBA', (WIDTH, HEIGHT), color)
+    return arcade.Texture(image)
 
 
-def remove_row(board, row):
-    """ Remove a row from the board, add a blank row on top. """
-    del board[row]
-    return [[0 for _ in range(COLUMN_COUNT)]] + board
+#
+# List of images per color in colors
+#
+texture_list = [create_texture(color) for color in colors]
 
 
-def join_matrixes(matrix_1, matrix_2, matrix_2_offset):
-    """ Copy matrix 2 onto matrix 1 based on the passed in x, y offset coordinate """
-    offset_x, offset_y = matrix_2_offset
-    for cy, row in enumerate(matrix_2):
-        for cx, val in enumerate(row):
-            matrix_1[cy + offset_y - 1][cx + offset_x] += val
-    return matrix_1
+class Board:
```

Comment:
This lengthens the code, significantly in what's supposed to be a simple how-to. Seem better for a longer example?